### PR TITLE
Add basic checks for bot activity on form

### PR DESCRIPTION
**Pull request by ccali11**
@shanejearley Here are the methods I've employed to deter bot sign-ups:
- **Standard email validation** using regex - needs to end in @something.something, for example
- **Hidden `<input>` "bot trap"** - some bots will fill out this input notifying us it is spam while real users will never see it
- **"Time analysis"** - rejects the email if submitted less than 3 seconds from landing on the web page
- **Email Address Length** - if shorter than 5 characters or longer than 100 characters, we reject the email

Check out [this article](https://www.loginradius.com/blog/identity/bot-attacks/) that I drew inspiration from and let me know if you think we should employ any of the other techniques. The other ones I'd consider are:
- **Adding a ReCaptcha** - although when I started down that path it looked like it was for GCP projects only; could be wrong about that. Will investigate further if you think this is worth the slight drawback to user experience. I think there are other "open source" options that we could potentially use as well.
- **Double Opt-In** - may be more trouble than it is worth, but open to doing this

The ones I didn't understand, which could be helpful to hear your interpretation of:
- **6. Hide Target Requests**
- **10. Beware of Cross-Site Request Forgery (CSRF)**

Submitting PR now (copying and pasting this comment from my comment made in issue #33 thread for your convenience)

### CODE REVIEW

File: apps/website/src/pages/index/index.vue [hunk 1]
```diff
@@ -10,13 +10,24 @@ const successMessage: Ref = ref<HTMLDivElement>()
 const invalidMessage: Ref = ref<HTMLDivElement>()
 
 const email = ref('')
+const botTrapInput = ref('')
+const timeArrived = ref(new Date().getTime())
+
 async function onSubmit() {
```
Comment (shanejearley) [inline apps/website/src/pages/index/index.vue:16]: Could you move all the bot logic (18-19 and 25-30) into `checkForBots` or some better name that you come up with?

File: apps/website/src/pages/index/index.vue [hunk 2]
```diff
@@ -10,13 +10,24 @@ const successMessage: Ref = ref<HTMLDivElement>()
...
+  if (isBotTrap) return
   if (!validEmail) {
     invalidMessage.value.style.display = 'block'
     return
   }
-  console.log('validEmail :>> ', validEmail)
+
+  const timeNow = new Date().getTime()
+  const timeDiff = timeNow - timeArrived.value
+  if (timeDiff < 3000) {
```
Comment (shanejearley) [inline apps/website/src/pages/index/index.vue:27]: Weeds out all those suspicious superusers too :-)

**Review by shanejearley — CHANGES_REQUESTED** [review]

**Comment by shanejearley** [comment]
> @shanejearley Here are the methods I've employed to deter bot sign-ups:
> 
> * **Standard email validation** using regex - needs to end in @something.something, for example
> * **Hidden `<input>` "bot trap"** - some bots will fill out this input notifying us it is spam while real users will never see it
> * **"Time analysis"** - rejects the email if submitted less than 3 seconds from landing on the web page
> * **Email Address Length** - if shorter than 5 characters or longer than 100 characters, we reject the email
> 
> Check out this article that I drew inspiration from and let me know if you think we should employ any of the other techniques. The other ones I'd consider are:

@ccali11 could you share the link? ^ Sounds good.

> 
> * **Adding a ReCaptcha** - although when I started down that path it looked like it was for GCP projects only; could be wrong about that. Will investigate further if you think this is worth the slight drawback to user experience. I think there are other "open source" options that we could potentially use as well.
> * **Double Opt-In** - may be more trouble than it is worth, but open to doing this
> 
> The ones I didn't understand, which could be helpful to hear your interpretation of:
> 
> * **6. Hide Target Requests**
> * **10. Beware of Cross-Site Request Forgery (CSRF)**
> 
> Submitting PR now (copying and pasting this comment from my comment made in issue #33 thread for your convenience)

**Comment by ccali11** [comment]
@shanejearley - I updated my initial PR comments with the article about ways to deter bots I referenced that I forgot to share link: https://github.com/consensusnetworks/casimir/pull/48#issuecomment-1167387821

File: apps/website/src/pages/index/index.vue [hunk 3]
```diff
@@ -111,64 +122,51 @@ function validateEmail(email: string) {
...
             alt=""
-          >
+          />
         </div>
         <div class="min-w-[375px] w-1/2 pt-[50px] pr-[50px]">
           <h1 class="text-[42px] text-left ">
             <span class="text-[#F36F38] font-extrabold">Manage</span> your
-            assets and earnings across chains. See all of your coins and NFTs
-            in one dashboard.
+            assets and earnings across chains. See all of your coins and NFTs in
```
[the remark this answers is not in the document]
Comment (ccali11) [reply apps/website/src/pages/index/index.vue:149]: eslint

File: apps/website/src/pages/index/index.vue [hunk 4]
```diff
@@ -111,64 +122,51 @@ function validateEmail(email: string) {
           >
             Please enter a valid email.
           </div>
-          <span class="small-text text-[#667085] pl-[5px]">We won't spam you. We promise.</span>
+          <span class="small-text text-[#667085] pl-[5px]"
```
[the remark this answers is not in the document]
Comment (ccali11) [reply apps/website/src/pages/index/index.vue]: done.

File: apps/website/src/pages/index/index.vue (hunk 1 above)
Comment (ccali11) [reply apps/website/src/pages/index/index.vue:16]: done.

**Pull request merged**